src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -149,6 +149,11 @@ namespace winrt::TerminalApp::implementation
         _languageProfileNotifier = winrt::make_self<LanguageProfileNotifier>([this]() {
             _reloadSettings->Run();
         });
+
+        // Do this here, rather than at the top of main. This will prevent us from
+        // including this variable in the vars we serialize in the
+        // Remoting::CommandlineArgs up in HandleCommandlineArgs.
+        _setupFolderPathEnvVar();

```

Comment:
FWIW an alternative, but more robust approach may be to have our own `ExpandEnvironmentStringsW`. It would also work better together with virtual env vars and the like. We already have such a function in a way in `til::env` already too.